frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachineTemplate.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable no-unused-vars, no-undef, no-await-in-loop, no-console */
+import { browser } from 'protractor';
+import { testName } from '../../../../../integration-tests/protractor.conf';
+import {
+  errorMessage,
+  filterForName,
+  resourceRowsPresent,
+} from '../../../../../integration-tests/views/crud.view';
+import { VMTemplateConfig } from '../utils/types';
+import { WIZARD_CREATE_TEMPLATE_ERROR, WIZARD_TABLE_FIRST_ROW } from '../utils/consts';
+import { Wizard } from './wizard';
+import { KubevirtDetailView } from './kubevirtDetailView';
+
+export class VirtualMachineTemplate extends KubevirtDetailView {
+  constructor(templateConfig) {
+    super({ ...templateConfig, kind: 'vmtemplates' });
+  }
+
+  async create({
+    name,
+    namespace,
+    description,
+    provisionSource,
+    operatingSystem,
+    flavor,
+    workloadProfile,
+    cloudInit,
+    storageResources,
+    networkResources,
+  }: VMTemplateConfig) {
+    await this.navigateToListView();
+
+    // Basic Settings for VM template
+    const wizard = new Wizard();
+    await wizard.openWizard();
+    await wizard.fillName(name);
+    await wizard.fillDescription(description);
+    if (!(await browser.getCurrentUrl()).includes(`${testName}/${this.kind}`)) {
+      await wizard.selectNamespace(namespace);
+    }
+    await wizard.selectProvisionSource(provisionSource);
+    await wizard.selectOperatingSystem(operatingSystem);
+    await wizard.selectFlavor(flavor);
+    await wizard.selectWorkloadProfile(workloadProfile);
+    if (cloudInit.useCloudInit) {
+      await wizard.useCloudInit(cloudInit);
+    }
+    await wizard.next();
+
+    // Networking
+    for (const networkOption of networkResources) {
+      await wizard.addNIC(
```

Comment:
doesn't this function expect the networkOption object instead?

Reply:
no using networkOption :tada:

Reply:
ah, I have mistaken that with `KubevirtDetailView.addNIC`. Thanks for fixing the interface